src/detect-pcre.c

```diff
@@ -199,7 +200,8 @@ int DetectPcrePayloadMatch(DetectEngineThreadCtx *det_ctx, const Signature *s,
     }
 
     /* run the actual pcre detection */
-    ret = DetectPcreExec(det_ctx, pe, (char *)ptr, len, start_offset, 0);
+    pcre2_match_data *match = pcre2_match_data_create_from_pattern(pe->parse_regex.regex, NULL);
```

Comment:
you can use thread specific data with `DetectRegisterThreadCtxFuncs`, `DetectThreadCtxGetKeywordThreadCtx`. See `src/detect-lua.c` for an example implementation.

Reply:
Thanks Victor.
I gathered from your answer that
- this was indeed one bug
- the performance impact could be avoided